functions/src/index.ts

```diff
@@ -232,6 +232,123 @@ function cleanAverages(averages: SensorReading[]): CleanedReadings {
   return new CleanedReadings(latitude, longitude, cleanedAverages);
 }
 
+/**
+ * Computes the AQI for PM 2.5 given the appropriate AQI breakpoints. in most use cases, this function will be called from a function which knows those breakpoints.
+ *
+ * Adapted from EPA function available at https://www.airnow.gov/sites/default/files/custom-js/conc-aqi.js
+ *
+ * @param pm25Concentration - the PM 2.5 concentration to calculate the AQI for
+ * @param lowConcentrationBreakpoint - the low breakpoint for PM 2.5 that the concentration falls within
+ * @param highConcentrationBreakpoint - the high breakpoint for PM 2.5 that the concentration falls within
+ * @param lowIndexBreakpoint - the low breakpoint for AQI that the AQI will fall between
+ * @param highIndexBreakpoint - the high breakpoint for AQI that the AQI will fall between
+ */
+function _indexCalculation(
+  highIndexBreakpoint: number,
+  lowIndexBreakpoint: number,
+  highConcentrationBreakpoint: number,
+  lowConcentrationBreakpoint: number,
+  pm25Concentration: number
+): number {
+  const indexRange = highIndexBreakpoint - lowIndexBreakpoint;
+  const concentrationRange =
+    highConcentrationBreakpoint - lowConcentrationBreakpoint;
+  const rangeRelativeConcentration =
+    pm25Concentration - lowConcentrationBreakpoint;
+
+  return (
+    (indexRange / concentrationRange) * rangeRelativeConcentration +
+    lowIndexBreakpoint
+  );
+}
+
+/**
+ * Computes the AQI for a given PM 2.5 concentration in micrograms per cubic meter.
+ *
+ * Adapted from EPA functions available at https://www.airnow.gov/sites/default/files/custom-js/conc-aqi.js
+ * @param pm25Conc - the PM 2.5 concentration to calculate the AQI for
+ * @returns the AQI for a given PM2.5 rounded to the nearest integer for valid AQI ranges. If the PM 2.5 concentration is less than zero, the reported AQI is negative infinity. If the PM 2.5 concentration is too high ("beyond the AQI" in EPA parlance), positive infinity is reported.
```

Comment:
Break this line up into multiple lines

Reply:
I'm not completely sure the best way to do so while maintaining formatting, but if possible, we should do so

Reply:
Upon some research: I say make the `@returns` description minimal. Then add a `@remarks` tag below that gives more details.